src/CrudPanel.php

```diff
@@ -237,16 +237,61 @@ public function sort($items)
      *          App/Models/Address defined by a company() method on the user model and an address() method on the
      *          company model, the 'App/Models/Address' string will be returned.
      *
-     * @param $relationString String Relation string. A dot notation can be used to chain multiple relations.
+     * @param string $relationString Relation string. A dot notation can be used to chain multiple relations.
+     * @param int $length Optionally specify a sub
+     * @param mixed $model Optionally specify a different model than the one in the crud object
      *
-     * @return String relation model name
+     * @return string relation model name
      */
-    private function getRelationModel($relationString)
+    public function getRelationModel($relationString, $length = null, $model = null)
     {
-        $result = array_reduce(explode(".", $relationString), function ($obj, $method) {
+        $relationArray = explode('.', $relationString);
+        if ($length == null) {
+            $length = count($relationArray);
+        }
+        if ($model == null) {
+            $model = $this->model;
+        }
+        $result = array_reduce(array_splice($relationArray, 0, $length), function ($obj, $method) {
             return $obj->$method()->getRelated();
-        }, $this->model);
+        }, $model);
 
         return get_class($result);
     }
+
+    /**
+     * Get the given attribute from the specified nested relations of the given model
+     * (ex: the list of streets from the many addresses of the company of a given user).
+     *
+     * @param mixed $model model (ex: user)
+     * @param string $relationString model relation string chained using the dot notation (ex: user.company.address)
+     * @param string $attribute the attribute from the chain relation model (ex: the street attribute from the address model)
+     * @param array $resultedValues result aggregator
+     *
+     * @return array an array containing a list of attributes from the given chained relation
+     */
+    public function getAttributeFromNestedRelations($model, $relationString, $attribute, &$resultedValues = [])
+    {
+        $relationArray = explode('.', $relationString);
+        if (count($relationArray) == 1 || get_class($model) == $this->getRelationModel($relationString, -1, $model)) {
+            if ($model->{$relationString}) {
```

Comment:
As a programmer of a statically type language myself, I find it odd too :).

But it's in PHP, so we might as well use it: http://us2.php.net/manual/en/language.variables.variable.php.

Laravel has it in lots of places:
https://github.com/illuminate/log/blob/master/Writer.php#L203
https://github.com/laravel/framework/blob/master/src/Illuminate/Auth/Access/Gate.php#L155
https://github.com/illuminate/database/blob/master/Schema/Blueprint.php#L150

etc.

How would you have done it differently?